internal/cmd/controller/gitops/reconciler/gitjob_controller.go

```diff
@@ -166,19 +166,28 @@ func (r *GitJobReconciler) createJob(ctx context.Context, gitRepo *v1alpha1.GitR
 		}
 		gitRepoFromCluster.Status.ObservedGeneration = gitRepoFromCluster.Generation
 
-		return r.Status().Update(ctx, &gitRepoFromCluster)
```

Comment:
I'm not convinced of this change. `Update` is immediately happening after `Get`, so it's weird that it finds a conflict (meaning another client wrote to the same object in the meantime).

In fact, I'm not sure this even correct, and there could be a race condition:

1. `createJob` receives a GitRepo, and does something based on that information
2. At the end, it updates the ObservedGeneration field in the status. However, the value being used is the one that was just read from the API, which may differ from the one used as an input.

The `Generation` field is used to represent a revision of the object `Spec` has changed. My understanding is that it `ObservedGeneration` should be set to the last revision being processed, so that if `ObservedGeneration != Generation`, it means that the a reconciliation is needed.

Reply:
Ah, good find. 
So, I think this is a left-over from when we had a separate gitjob resource.
When the `createJob` func returns, there is another function which updates the same status.
I think we can remove this.

That being said: 
* retryOnConflict only makes sense, when we use the outer value to update the freshly retrieved resource. It should be `gitRepoFromCluster.Status.ObservedGeneration = gitRepo.Generation`
* `gitRepoFromCluster` is a bit verbose, `t` is fine, the variable is not that important


Regarding this whole file:
* we might want to rename this reconciler to `GitOpsReconciler` and fix the comment which mentions a CronJobReconciler
* `newJob`, `computeJobSpec` and `generateInitContainer` do a good job at hiding what they actually create: containers for "fleet gitclone" and "fleet apply".
* actually this has a lot of overlap with our gitrepo reconciler in the other container, both update the same status. However both trigger on different events.